### PR TITLE
Add Bandwidth toxic

### PR DESCRIPTION
**Pull request by xthexder**
@Sirupsen @pushrax 

Depends on https://github.com/Shopify/toxiproxy/pull/47

Adds a bandwidth toxic for limiting connections to a certain bitrate.

### CODE REVIEW

File: io_chan.go [hunk 1]
```diff
@@ -1,20 +1,29 @@
 package main
 
-import "io"
+import (
+	"io"
+	"time"
+)
+
+// Simulates a TCP packet by storing a slice of bytes with the receive timestmap
```
Comment (Sirupsen) [inline io_chan.go]: TCP packet doesn't really make sense, rather you're buffering a stream—but I suppose that's what you mean?

**Comment by Sirupsen** [comment]
Testing this I can imagine is a little tough, but it shouldn't be impossible

**Comment by xthexder** [comment]
Working on a go benchmark/test for this right now. I've tested this manually with curl already and it works well.

File: io_chan.go (hunk 1 above)
Comment (xthexder) [inline io_chan.go]: This is actually part of https://github.com/Shopify/toxiproxy/pull/47, just the second commit is for this PR.

Comment (pushrax) [inline io_chan.go]: Comment still stands though.

**Comment by pushrax** [comment]
Could write a test that sends sentinel data over the connection and checks every few quanta how much data has been received, not sure if it is necessary though.

**Comment by xthexder** [comment]
The bandwidth toxic seems to be accurate from 1KB/s all the way up to 200MB/s (> 1gbit), after that it starts to get limited by the actual speed of the proxy.

Depending on the size of the writes going in, the noop benchmark produces some interesting numbers:
```
GOMAXPROCS=4
1kb writes: 70MB/s
10kb writes: 500MB/s
1mb writes: 680MB/s
10mb writes: 910MB/s
100mb writes: 800MB/s
```

I think the lower write sizes are actually being limited by the speed of the test suite, not the proxy.

**Comment by xthexder** [comment]
Just added a latency test as well, I'm getting 100-150µs when I run it locally, which I think is pretty good.

**Comment by Sirupsen** [comment]
Can you add the benchmark results to the FAQ and add this to the CHANGELOG? :)

**Comment by xthexder** [comment]
I'm not really sure what to write in the FAQ anymore, Just ran the same benchmarks on my desktop at home and got 20µs latency and a crazy 2100MB/s peak (100mb writes)

**Pull request merged**